services/welding-machine-defect-detection-model-service/app/models/model_loader.py

```
@@ -0,0 +1,41 @@
+
+
+def load_threshold(signal_type: str):
+    try:
+        with open("static/thresholds/welding_thresholds.json", "r") as f:
+            thresholds = json.load(f)
+            if signal_type not in thresholds:
+                raise KeyError(
+                    f"Threshold for signal type '{signal_type}' not found in configuration")
+            return thresholds[signal_type]
+    except FileNotFoundError:
+        raise FileNotFoundError("Threshold configuration file not found")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)